### PR TITLE
#21 settings page

### LINKED ISSUE #21: Creat a settings page
There should be a settings option in the dropdown menu that is displayed when clicking on username on navbar

The settings page should have a sidebar for different sections, keep 2 sections for  now profipe and account

On the right of the sidebar current section should be displayed, for now only complete these two sections profile and account

Profile section should display user's details in editable text fields and there should be save option at the  bottom for saving changes made in the details, on clicking you should display a confirmation dialog asking if you want to save the changes (like logout dialog)

You need not implement the saving functionality, just complete the frontend

For account section there should be option to change email address and change password, complete the frontend part only

### PR DESCRIPTION
**Pull request by abhinavmalhotra01**
tried to resolve #21 by adding settings page

## Description

added the required pages and components for settings page

## Issue Resolved

tried to resolve #21 

## Screenshots

Screenshot 2023-02-20 192249
Screenshot 2023-02-20 192548
Screenshot 2023-02-20 192600

## Checklist

Please make sure to review the following before submitting your PR:

- [ ✅] I have read the [contribution guidelines](https://github.com/ken1000minus7/Bloggy/blob/master/README.md).
- [ ✅] I have read the code of conduct.
- [✅ ] I have reviewed my submission in detail.

### CODE REVIEW

File: src/frontend/src/components/SettingSidebarItem.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    useEffect(()=>{
+        const handleWidth = ()=>{
+            setWidth(window.innerWidth)
+        }
+        window.addEventListener("resize",handleWidth)
+        return ()=>{
+            window.removeEventListener("resize",handleWidth)
+        }
+    })
```
Comment (ken1000minus7) [inline src/frontend/src/components/SettingSidebarItem.js]:
```suggestion
    const {width} = useWindowSize()
```
You can simply access the width using this hook now

File: src/frontend/src/index.css [hunk 2]
```diff
@@ -57,7 +57,6 @@
         display: block;
         list-style-type: disc;
         margin: 1em 0 1em;
-        padding-left: 40px;
```
Comment (ken1000minus7) [inline src/frontend/src/index.css]: Do not remove this, instead u can override the padding where u are using the element

**Review by ken1000minus7 — CHANGES_REQUESTED** [review]

File: src/frontend/src/index.css (hunk 2 above)
Comment (abhinavmalhotra01) [reply src/frontend/src/index.css]: changed sir

File: src/frontend/src/components/SettingSidebar.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import React, {useEffect, useState} from 'react'
+import {Link} from "react-router-dom";
+import { useLocation } from 'react-router-dom';
+import AccountCircleIcon from '@mui/icons-material/AccountCircle';
+import SettingsIcon from '@mui/icons-material/Settings';
+import { SidebarItem } from './SettingSidebarItem';
+
+const SettingSidebar = () => {
+    const [width,setWidth] = useState(window.innerWidth)
```
Comment (ken1000minus7) [inline src/frontend/src/components/SettingSidebar.js]: Change the width here as well

**Review by ken1000minus7 — APPROVED** [review]
Lgtm

**Pull request merged**